tests/unit/spec/base/mozilla-utils.js

```diff
@@ -9,6 +9,67 @@ describe('mozilla-utils.js', function() {
 
     'use strict';
 
+    describe('trans', function () {
+        var stringDiv;
+
+        beforeEach(function () {
+            stringDiv = '<div id="strings" data-global-close="Close" ' +
+            'data-global-next="Next" ' +
+            'data-global-previous="Previous"> ' +
+            '</div>';
+
+            var container = document.createElement('div');
+            container.innerHTML = stringDiv;
+            $(container).appendTo('body');
+        });
+
+        afterEach(function() {
+            $(stringDiv).remove();
+        });
+
+        it('should correctly return translation value', function () {
+            var translation = Mozilla.Utils.trans('global-next');
+            expect(translation === 'Next');
+        });
+    });
+
+    describe('initLangSwitcher', function () {
+        var form;
+
+        beforeEach(function () {
+            form = '<form id="lang_form" method="get" action="#">' +
+              '<label for="page-language-select">Langue</label>' +
+              '<select id="page-language-select" name="lang" dir="ltr">' +
+                  '<option lang="de" value="de">Deutsch</option>' +
+                  '<option lang="en-US" value="en-US" selected>English</option>' +
+                  '<option lang="fr" value="fr">Français</option>' +
+              '</select>' +
+              '<button type="submit">Go</button>' +
+            '</form>';
+
+            var container = document.createElement('div');
+            container.innerHTML = form;
+            $(container).appendTo('body');
+        });
+
+        afterEach(function() {
+            $(form).remove();
+        });
+
+        it('should correctly identify the previous and new languages', function () {
+            spyOn(Mozilla.Utils, 'doRedirect');
+            var $select = $('#page-language-select');
+            // init
+            Mozilla.Utils.initLangSwitcher();
+            // change the drop down
+            $select.val('de');
+            // fire the event
+            $select.trigger('change');
+            // check redirected to expected page
+            expect(location.href === 'http://localhost:9876/debug.html?lang=de#&step=1');
```

Comment:
Instead of asserting `location.href` I think it may be enough to just assert that `Utils.doRedirect()` was called? I'm not sure if we can really depend on the localhost address of the test runner to assert against reliably. If it may help, there are some equivalent tests for this feature over in the [protocol repo](https://github.com/mozilla/protocol/blob/master/tests/unit/protocol-lang-switcher.js#L67-L80) if you would like some plain JS examples for reference.

Reply:
I'm really trying to check that doRedirect is passed the correct parameters but it's passed something funny and I wasn't able to check it.

Reply:
And you're probably right, we can't rely on the host :/

Reply:
Updated to just check window.location.search. Better?